src/pages/index.js

```diff
@@ -93,11 +158,48 @@ cardPreviewModal.setEventListeners();
 const createCard = (data) => {
   const cardEl = new Card(
     {
+      id: data._id,
+      isLiked: data.isLiked,
       name: data.name,
       link: data.link,
       handleImageClick: (imgData) => {
         cardPreviewModal.open(imgData);
       },
+
+      handleRemoveCardClick: () => {
+        const id = cardEl.getId();
+        newRemoveCardModal.open();
+
+        newRemoveCardModal.setSubmitAction(() => {
+          newRemoveCardModal.setLoadingText(true);
+          api.deleteCard(id).then((res) => {
+            cardEl.handleRemoveCard(res);
+
+          }) .catch((err) => {
+            console.error(err);
+          })
+          .finally(() => {
+            newRemoveCardModal.setLoadingText(false);
+          });
+        });
+      },
+      handleLikeIcon: (card) => {
+        if (!card.getLikeStatus()) {
+          return api
+            .addLike(card._id)
+            .then(() => {
+              card.getLikeStatus(true);
+            })
+            .catch(console.err);
+        } else {
+          return api
+            .removeLike(card._id)
+            .then(() => {
+              card.getLikeStatus(false);
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

`getLikeStatus` doesn't have any effect


Please, move the code of **toggling** into `setLikeStatus` and call this method in `then` here

```js
              card.setLikeStatus(false);


```